models/filter/model_filter.go

```diff
@@ -0,0 +1,23 @@
+package filter
+
+// ModelFilter is a common model filter that contains common fields for all model filters.
+type ModelFilter struct {
+	IncludeDeleted *bool      `json:"include_deleted,omitempty" example:"true"`
+	CreatedRange   *TimeRange `json:"created_range,omitempty" swaggertype:"object,string"`
+	UpdatedRange   *TimeRange `json:"updated_range,omitempty" swaggertype:"object,string"`
+}
+
+// ToDbConditions converts filter fields to the datastore conditions using gorm naming strategy
+func (mf *ModelFilter) ToDbConditions() map[string]interface{} {
+	conditions := map[string]interface{}{}
+
+	applyConditionsIfNotNil(conditions, "created_at", mf.CreatedRange.ToDbConditions())
+	applyConditionsIfNotNil(conditions, "updated_at", mf.UpdatedRange.ToDbConditions())
+
+	if mf.IncludeDeleted == nil || !*mf.IncludeDeleted {
```

Comment:
If we have 2 records. One active and second is deleted. With this condition we can get one of them or both ?

Reply:
As the `IncludeDeleted` field says:
When we have `deleted-record` and `normal-record`
- when `IncludeDeleted` is set to true - we'll get `deleted-record` and `normal-record`
- when `IncludeDeleted` is nil or is set to false - we'll get only `normal-record`